Sources/Common/DataModel/ImageData/index.js

```diff
@@ -319,8 +309,8 @@ function vtkImageData(publicAPI, model) {
 
   // slow version for generic arrays
   publicAPI.indexToWorld = (ain, aout = []) => {
-    const vin = vec3.fromValues(ain[0], ain[1], ain[2]);
-    const vout = vec3.create();
+    const vin = new Float64Array([ain[0], ain[1], ain[2]]);
```

Comment:
Float64Array.from(ain)

Reply:
Only if I know ain is of length 3, in cases where I knew that I did it. In cases like this I wasn't ready to assume it was a length 3 vector

Reply:
I do not understand the purpose of this function (or to be more precise, the purpose of `worldToIndexVec3`).
shouldn't it be simply
```
publicAPI.indexToWorld = (ain, aout = []) => {
 vec3.transformMat4(aout , ain, model.indexToWorld);
}
```
Javascript numbers are 64 bits by default, no ? Even on 32 bit machines, no? So that means that regular `Array`s are 64bits.

Why bother creating intermediate 64 bit arrays ?

Reply:
@jourdain @finetjul I did another pass and improved about another 50 places that were doing more allocs than needed.  If you can please take a look to check that I didn't do something wrong. There are a ton of other places in vtk.js that can be improved in terms of heap allocs etc.

More broadly I'm not 100% sure why we are using Float64 at all for these small arrays (3,4,9,16). WebGL takes Float32 so we are converting form Float64. We definitely want double calcs everywhere but Float64 and JS numbers are both double. From what I can find there is some belief that native arrays can be optimized more easily than generic arrays, but other than that I'm not sure if there is any difference for these small vector/matrices.

Reply:
I briefly looked, it looks fine to me. Thanks !

I think we should decide of a convention: when should we instantiate Float64Array when a JS array?

I'm surprised you did not run into troubles with `macro.setArray`, it does not support well typed arrays.

Reply:
There is actually a small issue: `vtkImageData.indexToWorldVec3` is now doing the same than `vtkImageData.indexToWorld`. One function must be removed. Same with `worldToIndex`.

Reply:
There were a number of methods that had

const foo = Float64Array(3); 
...
return [[foo[0], foo[1], foo[2]];

that I changed to just return foo. I noticed that done in a few other places. That worried me a bit but it all seemed to work and I'm not sure why it wouldn't as my knowledge of JS type issues is weak. 

WRT Array versus Float64Array from what I can see TypedArrays are the fastest option, BUT the JS compilers are smart and as long as you keep a JS array homogeneous it basically makes it a TypedArray under the hood. My gut is that if we know an array should only be used for storing one type, we should explicitly create it as that type. But interested in what @jourdain is thinking as well.

Reply:
returning foo directly make sense. 

I share your opinion of Use TypedArray when `used for storing one type` and the size is known.

Reply:
Except maybe when you already know its content like `[0.5, 1, -1]`. I would not convert that to `new Float64Array([0.5, 1, -1])`.

Reply:
Yup, thanks!